Cargo.toml

```diff
@@ -101,24 +100,48 @@ lto = true
 [workspace.dependencies]
 bytes = { version = "1" }
 drain = { version = "0.1", default-features = false }
-h2 = { version = "0.3" }
-http = { version = "0.2" }
-http-body = { version = "0.4" }
-hyper = { version = "0.14.32", default-features = false }
-prost = { version = "0.12" }
-prost-build = { version = "0.12", default-features = false }
-prost-types = { version = "0.12" }
-tokio-rustls = { version = "0.26", default-features = false, features = [
-    "ring",
-    "logging",
-] }
-tonic = { version = "0.10", default-features = false }
-tonic-build = { version = "0.10", default-features = false }
-tower = { version = "0.4.13", default-features = false }
+h2 = { version = "0.4" }
+http = { version = "1" }
+http-body = { version = "1" }
+hyper = { version = "1", default-features = false }
+hyper-util = { version = "0.1", default-features = false }
+prost = { version = "0.13" }
+prost-build = { version = "0.13", default-features = false }
+prost-types = { version = "0.13" }
+tonic = { version = "0.12", default-features = false }
+tonic-build = { version = "0.12", default-features = false }
+tower = { version = "0.4", default-features = false }
```

Comment:
Do we want tower 0.5? We're pulling in dupes currently. My understanding is that it's part of the tonic bump, effectively. I suppose if we're compiling as it is, we can defer this as a follow-up.

Reply:
i'd like to defer this into a follow-up, which i'll plan to handle in parallel to review for this change.

https://github.com/tower-rs/tower/pull/637, undocumented in the 0.5 changelog, introduced breaking changes that are sufficiently involved enough for me to skew towards handling them separately.

Reply:
for posterity, #3744 contains some of the initial steps towards upgrading `tower`. at the time of writing, that pull request is still a draft, but should position us to land as a quick follow-on to this work.

tower-rs/tower#703 is another change that's somewhat relevant to us as a user of the `Policy` ~struct~ trait.